### PR TITLE
re: Rename active peer to connected peer terminology in `near-network`

### PR DESCRIPTION
**Pull request by pmnoxx**
Change terminology used in `near-network` to be more accurate.

Currently we use `active peer` terminology to mean `connected peer`.
Unfortunately, that term is not self explanatory and leads to some confussion.

See #5424

Blocked by #5515

### CODE REVIEW

File: chain/client-primitives/src/types.rs
Comment (frol) [inline chain/client-primitives/src/types.rs:468]: This will break JSON RPC `network` backward compatibility since the old names won't be there:

```
$ http post https://rpc.testnet.near.org jsonrpc=2.0 method=network_info params:='[]' id=dontcare
```

It is suboptimal to expect this type to be serializable straight to JSON RPC response, so I suggest we go and fix it in https://github.com/near/nearcore/blob/9e6c2078dae67938a4fa29a13bb502b04819b7b8/chain/jsonrpc-primitives/src/types/network_info.rs#L5-L6, so instead of flattening `NetworkInfoResponse` there, let's have a copy of the old fields over:

```rust
    pub active_peers: Vec<PeerInfo>,
    pub num_active_peers: usize,
    pub peer_max_count: u32,
    pub sent_bytes_per_sec: u64,
    pub received_bytes_per_sec: u64,
    /// Accounts of known block and chunk producers from routing table.
    pub known_producers: Vec<KnownProducer>,
```

 and implement the `From`:

https://github.com/near/nearcore/blob/9e6c2078dae67938a4fa29a13bb502b04819b7b8/chain/jsonrpc-primitives/src/types/network_info.rs#L16-L20

**Review by frol — APPROVED** [review]
I am approving to unblock the PR, but it is important to address the backward compatibility comment

File: chain/client-primitives/src/types.rs
Comment (mm-near) [reply chain/client-primitives/src/types.rs:468]: That's a very good catch.. And a gap in our testing (as we should have a test that loads the JSON requests from a file - to simulate backwards compatiblity).

**Review by mm-near — DISMISSED** [review]
let's make sure that backwards compatibility is fixed before proceeding with this PR

File: chain/client-primitives/src/types.rs
Comment (pmnoxx) [reply chain/client-primitives/src/types.rs:468]: We should do the code separation. I'm changing this PR to `draft` until we merge https://github.com/near/nearcore/pull/5515

I created a separate issue for `jsonrpc`  https://github.com/near/nearcore/issues/5516

**Review by miraclx — APPROVED** [review]

**Review by chefsale — APPROVED** [review]

**Review by mm-near — APPROVED** [review]

**Pull request merged**
